### PR TITLE
f/qcomponent

### PR DESCRIPTION
**Pull request by whittlbc**
Added a `QComponent` abstract class that extends the default React `Component`. Can be used to easily inherit methods/properties that should be shared across multiple components. Currently using it to inherit the `MOBILE_THRESH` property, which specifies the px width at which our app converts to a mobile layout.

### CODE REVIEW

**Comment by atn34** [comment]
Looks good. I am confused by the merge commit where you merged master into a feature branch. Wouldn't it make more sense to rebase the feature branch onto master?

**Comment by whittlbc** [comment]
Yeah that happened cause I decided to commit directly to master for [this](https://github.com/quokkateam/quokka/commit/91fd2fcff6479156a9dc7370c6468d6035d3ebd6) rather than keeping it in this branch. Then I switched back to this branch and `git pull`ed from master. So that's why that happened.

File: src/components/abstract/QComponent.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React, { Component } from 'react';
```
Comment (atn34) [inline src/components/abstract/QComponent.js]: React is not used. (This was probably copied from a js file that uses jsx like `<Component />`, which requires `React` to be in scope)

Comment (whittlbc) [reply src/components/abstract/QComponent.js]: Ahaaa. Interesting. Good to know. Will remove.

**Pull request merged**